### ISSUE TITLE
Broken IE11 compatibility (Babel?)

### ISSUE THREAD

**Issue by coderars** [andreknieriem/simplelightbox#235]
Hi @andreknieriem ! Thank you for this great plugin!

The list of features mentions:

> Works in every modern Browser, even in IE 11

currently, **for IE11 this is not true**. It fails even on the [demo site](https://simplelightbox.com/) with error:

> Object doesn't support property or method "assign"

Running it through my project's Babel config (with Object assign polyfill) this error disappears, but another one comes up:

> Object doesn't support this action

at [this line](https://github.com/andreknieriem/simplelightbox/blob/731bf85475ae95c5e564f24c9d6365bc7c17483e/src/simple-lightbox.js#L1081) of the code:

````js
element.dispatchEvent(new Event('show.' + this.eventNamespace));
````

Maybe [this issue](https://github.com/zloirock/core-js/issues/354) @ `core-js` could be related (not sure).

````js
// Please do not drop IE11 compatibility 😃
// I have to support it 😭 and I really like your plugin (thanks again!) 
````

**Comment by andreknieriem** [comment]
Hey, did you tried the simplelightbox/dist/simple-lightbox.legacy.min.js version? This is for old Browsers without es6 features. Best regards

**Comment by coderars** [comment]
Yes, meanwhile I've noticed that file, but I use Webpack with my own Babel config and polyfills are based on `useBuiltIns: "usage"` , so using that file is not an option for me. I'm afraid it'll be my personal fight with Babel 👿 to make this work with IE11. Sorry for the false report, please close this issue.

**Comment by coderars** [comment]
However, looking at your `legacy.js` file, it begins with

````js
import "core-js/stable";
import "regenerator-runtime/runtime";
````

and AFAIK this means that Babel will import **ALL polyfills**, not just those that are really needed for your plugin. This is why the bundled `simple-lightbox.legacy.js` is so huge... If you set `useBuiltIns: "usage"` in your `.babelrc` then those manual imports are not necessary.

**Comment by coderars** [comment]
Finally, I've succeeded to make it work on IE11 with my own Babel config and compile this plugin into my vendors.js. However, a little modification is needed in the code of `simple-lightbox.js` because [Explorer dies](https://stackoverflow.com/questions/26596123/internet-explorer-9-10-11-event-constructor-doesnt-work) on `new Event('myeventname')`. Unfortunately, Babel [won't fix this](https://github.com/zloirock/core-js/issues/354), but the solution is not so complicated and is totally safe.

So my suggested fix in `simple-lightbox.js` is to add a new method to `class SimpleLightbox` for creating the events to make it cross-browser:
````js
    createNewEvent(eventName) {
        let event;

        if (typeof(Event) === 'function') {
            // the standard way
            event = new Event(eventName);
        } else {
            // the IE way
            event = document.createEvent('Event');
            event.initEvent(eventName, true, true);
        }

        return event;
    }
````
then replace all the `new Event` constructors to `createNewEvent` like:
````js
element.dispatchEvent(this.createNewEvent('close.simplelightbox'));
````
I've tested and it works!

Alternative solution:
Another solution would be to replace `new Event` with `new CustomEvent` but that would require the end-user to manually include a polyfill like [custom-event-polyfill](https://www.npmjs.com/package/custom-event-polyfill) or [this script](https://stackoverflow.com/a/26596324/5263363). I've also tested it and it works too, but I think the first one is far better.
